modules/util/source-formatter/src/test/java/com/liferay/source/formatter/processor/UpgradeCatchAllCheckTest.java

```diff
@@ -0,0 +1,160 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.processor;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.json.JSONArrayImpl;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.util.ArrayUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.SourceFormatterArgs;
+import com.liferay.source.formatter.check.UpgradeCatchAllCheck;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+/**
+ * @author Kyle Miho
+ */
+@RunWith(Parameterized.class)
+public class UpgradeCatchAllCheckTest extends BaseSourceProcessorTestCase {
+
+	@Parameterized.Parameters(name = "Testcase-{index}: Testing {0} .{1}")
+	public static Iterable<Object[]> data() throws Exception {
+		List<Object[]> tickets = new ArrayList<>();
+
+		String issueKeysProp = System.getProperty("issue.key", null);
+
+		String[] issueKeys = StringUtil.split(issueKeysProp, StringPool.COMMA);
+
+		ClassLoader classLoader = UpgradeCatchAllCheck.class.getClassLoader();
+
+		JSONArray jsonArray = new JSONArrayImpl(
+			StringUtil.read(
+				classLoader.getResourceAsStream(
+					"dependencies/replacements.json")));
```

Comment:
@drewbrokke fixed, fyi I need to initialize JSONFactoryUtil during test paramterization, which occurs before setup, so i had to add it within the same function.

Reply:
Can it be done during `setUpClass` [like this](https://github.com/liferay/liferay-portal/blob/master/modules/util/client-extension-util-spring-boot/src/test/java/com/liferay/client/extension/util/spring/boot/LiferayOAuth2ResourceServerEnableWebSecurityTest.java?plain=1#L61)?

Reply:
@drewbrokke unfortunately not, since data() for paramterized test cases is executed to determine what tests need to be run, it also happens to run before both setup and setUpClass, so if we depend on using JSONFactoryUtil we have to run it within its own function